src/server/pfs/server/driver.go

```diff
@@ -987,188 +880,99 @@ func (d *driver) repoSize(ctx context.Context, repo *pfs.Repo) (int64, error) {
-		); err != nil && !col.IsErrNotFound(err) {
-			return errors.EnsureStack(err)
-		} else if err == nil {
-			if len(subvBI.DirectProvenance) != len(oldCommit.DirectProvenance) {
-				return errors.EnsureStack(pfsserver.ErrInconsistentCommit{Branch: subvBI.Branch, Commit: oldCommit.Commit})
-			}
-			for _, br := range oldCommit.DirectProvenance {
-				if !has(&subvBI.DirectProvenance, br) {
-					return errors.EnsureStack(pfsserver.ErrInconsistentCommit{Branch: subvBI.Branch, Commit: oldCommit.Commit})
+		newCommit := &pfs.Commit{
+			Repo:   bi.Branch.Repo,
+			Branch: bi.Branch,
+			ID:     txnCtx.CommitSetID,
+		}
+		newCommitInfo := &pfs.CommitInfo{
+			Commit:  newCommit,
+			Origin:  &pfs.CommitOrigin{Kind: pfs.OriginKind_AUTO},
+			Started: txnCtx.Timestamp,
+		}
+		// enumerate the new commit's provenance
+		for _, b := range bi.DirectProvenance {
+			var provCommit *pfs.Commit
+			if pbi, ok := seen[pfsdb.BranchKey(b)]; ok {
+				provCommit = client.NewProjectCommit(pbi.Branch.Repo.Project.Name, pbi.Branch.Repo.Name, pbi.Branch.Name, txnCtx.CommitSetID)
+			} else {
+				provBranchInfo := &pfs.BranchInfo{}
+				if err := d.branches.ReadWrite(txnCtx.SqlTx).Get(pfsdb.BranchKey(b), provBranchInfo); err != nil {
+					return errors.Wrapf(err, "get provenant branch %q", pfsdb.BranchKey(b))
 				}
-			}
-			// the old commit is compatible with the current provenance, so use it.
-			// This will reuse the old data and not create a job, meaning if the reprocess spec is "every job",
-			// moving a branch head back is different from doing the inverse changes in PFS
-			subvBI.Head = oldCommit.Commit
-			if err := d.branches.ReadWrite(txnCtx.SqlTx).Put(subvBI.Branch, subvBI); err != nil {
-				return errors.EnsureStack(err)
-			}
-		} else {
-			// This branch has no commit for this CommitSet, start a new output commit in 'subvBI.Branch'
-			newCommit := &pfs.Commit{
-				Branch: subvBI.Branch,
-				ID:     txnCtx.CommitSetID,
-			}
-			newCommitInfo := &pfs.CommitInfo{
-				Commit:           newCommit,
-				Origin:           &pfs.CommitOrigin{Kind: pfs.OriginKind_AUTO},
-				Started:          txnCtx.Timestamp,
-				DirectProvenance: subvBI.DirectProvenance,
-			}
-
-			// we might be able to find an older parent commit that better reflects the provenance state, saving work
-
-			// Set 'newCommit's ParentCommit, 'branch.Head's ChildCommits and 'branch.Head'
-			newCommitInfo.ParentCommit = subvBI.Head
-			subvBI.Head = newCommit
-			if newCommitInfo.ParentCommit != nil {
-				parentCommitInfo := &pfs.CommitInfo{}
-				if err := d.commits.ReadWrite(txnCtx.SqlTx).Update(newCommitInfo.ParentCommit, parentCommitInfo, func() error {
+				seenAliases[pfsdb.BranchKey(b)] = provBranchInfo
+				provCommit = provBranchInfo.Head
+			}
+			newCommitInfo.DirectProvenance = append(newCommitInfo.DirectProvenance, provCommit)
+		}
+		// we might be able to find an older parent commit that better reflects the provenance state, saving work
```

Comment:
What is this referring to?

Reply:
not really sure, I'll remove this. I think I was commenting on selecting a parent commit to support incrementally more optimally